### PR TITLE
[ticket/17321] Provide correct push worker update

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

<a href="https://tracker.phpbb.com/browse/PHPBB3-17321">PHPBB3-17321</a>.

### CODE REVIEW

File: phpBB/assets/javascript/update_worker.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		} else {
+			callBack();
+		}
+	}
+}
+
+domReady(() => {
+	/* global serviceWorkerUrl */
+	webpushWorkerUpdate();
+});
```
Comment (iMattPro) [inline phpBB/assets/javascript/update_worker.js:29]: Try this:
```
'use strict';

function webpushWorkerUpdate() {
	if ('serviceWorker' in navigator) {
		navigator.serviceWorker.getRegistration(serviceWorkerUrl)
			.then(registration => {
				registration.update();
			})
			.catch(error => {
				// Service worker could not be updated
				console.info(error);
			});
	}
}

// Do not redeclare function if exist
if (typeof domReady === 'undefined') {
	window.domReady = function(callBack) {
		if (document.readyState === 'loading') {
			document.addEventListener('DOMContentLoaded', callBack);
		} else {
			callBack();
		}
	};
}

/* global domReady */
domReady(() => {
	/* global serviceWorkerUrl */
	webpushWorkerUpdate();
});
```

File: phpBB/assets/javascript/update_worker.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	window.domReady = function(callBack) {
+		if (document.readyState === 'loading') {
+			document.addEventListener('DOMContentLoaded', callBack);
+		} else {
+			callBack();
+		}
+	};
+}
+
+/* global domReady */
```
Comment (iMattPro) [inline phpBB/assets/javascript/update_worker.js:27]: @marc1706 do you know why stuff like this `domReady` failed linting here w/o having to make that global content, but when you had the same code in your webpush.js file, that passed linting no problem?

Comment (marc1706) [reply phpBB/assets/javascript/update_worker.js:27]: magic

**Comment by marc1706** [comment]
@rxu can you check out this alternative? https://github.com/phpbb/phpbb/pull/6631

**Comment by rxu** [comment]
Closing in favour of #6631.

**Pull request closed without merging**